app/api.py

```diff
@@ -339,19 +339,63 @@ def create_price(
         else:
             # Check if the proof belongs to the current user
             # Only proof uploaded by the user can be used
-            if proof.owner != current_user.user_id:
+            if db_proof.owner != current_user.user_id:
                 raise HTTPException(
                     status_code=status.HTTP_403_FORBIDDEN,
                     detail="Proof does not belong to current user",
                 )
-
+    # create price
     db_price = crud.create_price(db, price=price, user=current_user)
+    # update counts
     background_tasks.add_task(tasks.create_price_product, db, price=db_price)
     background_tasks.add_task(tasks.create_price_location, db, price=db_price)
     background_tasks.add_task(tasks.increment_user_price_count, db, user=current_user)
     return db_price
 
 
+@app.delete(
+    "/api/v1/prices/{price_id}",
+    status_code=status.HTTP_204_NO_CONTENT,
+    tags=["Prices"],
+)
+def delete_price(
+    price_id: int,
+    background_tasks: BackgroundTasks,
+    current_user: schemas.UserCreate = Depends(get_current_user),
+    db: Session = Depends(get_db),
+):
+    """
+    Delete a price.
+
+    This endpoint requires authentication.
+    A user can delete only owned prices.
+    """
+    db_price = crud.get_price_by_id(db, price_id=price_id)
+    # get price
+    if not db_price:
+        raise HTTPException(
+            status_code=404,
+            detail=f"Price with code {price_id} not found",
+        )
+    # Check if the price belongs to the current user
+    if db_price.owner != current_user.user_id:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="Price does not belong to current user",
+        )
+    # update counts
+    background_tasks.add_task(tasks.decrement_user_price_count, db, user=current_user)
```

Comment:
Why do you do this as a background task? I would rather have it in the endpoint, so that the full operation is atomic.

Reply:
ok fixed ! i removed the background tasks, and did 1 single crud method / commit instead :)